script/check-posix-build-cmake

```diff
@@ -76,7 +76,7 @@ main()
     fi
 
     reset_source
-    build -DOT_POSIX_CONFIG_RCP_BUS=VENDOR "$@"
+    build -DOT_POSIX_RCP_VENDOR_BUS=ON "$@"
```

Comment:
If we are renaming this, let's add a check in `cmake` to give error if the old option `OT_POSIX_CONFIG_RCP_BUS` is being used by a project/build.

(we can do something similar to what we do in `options.cmake` -> https://github.com/openthread/openthread/blob/c6bece0729f0c962766ccbf01d2178e984c9f8d4/etc/cmake/options.cmake#L343-L356)